packages/ui/src/models/visualization/flows/abstract-camel-visual-entity.ts

```diff
@@ -291,6 +291,18 @@ export abstract class AbstractCamelVisualEntity<T extends object> implements Bas
     fromNode.getChildren()?.splice(0);
     fromNode.data.isGroup = false;
 
+    const normalizeGroups = (node: IVisualizationNode) => {
+      const children = node.getChildren() ?? [];
+
+      if (node.data.isGroup && children.length === 0) {
+        node.data.isGroup = false;
+      }
+
+      children.forEach((child) => normalizeGroups(child));
+    };
```

Comment:
💅: We could extract this function to a dedicated file in the [`utils`](https://github.com/KaotoIO/kaoto/tree/dc7dc31d3b024292ad0cdb352e8bfa0b3f7d8c08/packages/ui/src/utils) folder so we can add dedicated unit tests for it.